pkg/deploy/api/v1/types.go

```diff
@@ -343,3 +343,24 @@ type DeploymentConfigRollbackSpec struct {
 	// IncludeStrategy specifies whether to include the deployment Strategy.
 	IncludeStrategy bool `json:"includeStrategy" description:"whether to include the deployment strategy in the rollback"`
 }
+
+// DeploymentLog represents the logs for a deployment
+type DeploymentLog struct {
+	kapi.TypeMeta `json:",inline"`
+}
+
+// DeploymentLogOptions is the REST options for a deployment log
+type DeploymentLogOptions struct {
+	kapi.TypeMeta
+
+	// Follow if true indicates that the deployment log should be streamed until
+	// the deployment terminates.
+	Follow bool `json:"follow,omitempty" description:"if true indicates that the log should be streamed; defaults to false"`
+
+	// NoWait if true causes the call to return immediately even if the deployment
+	// is not available yet. Otherwise the server will wait until the deployment has started.
+	NoWait bool `json:"nowait,omitempty" description:"if true indicates that the server should not wait for a log to be available before returning; defaults to false"`
```

Comment:
the JSON for this should be `noWait` - if build log is like that we can stay consistent (but it's another thing we'll need to fix in V2, so add a TODO)